packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -889,7 +890,8 @@ export default abstract class SqlIntegration
   }
 
   getMetricAnalysisQuery(params: MetricAnalysisParams): string {
-    const { metric, settings } = params;
+    const metric = getMetricWithFiltersApplied(params);
+    const { settings } = params;
```

Comment:
tiny tiny nit: it might be nice if it was impossible to get metric from `params` directly and instead have to pass it through the validator above....

Related, it might be better to do this directly in the MetricAnalysisQueryRunner. It makes stuff like this feel a bit better, and it's moving a way from a pattern we use too much in SqlIntegration. I feel like these methods have too much override type logic that should be handled by the runner.